test/integration/test_convention_architecture.py

```diff
@@ -0,0 +1,95 @@
+"""
+Tests for the new convention architecture.
+"""
+import numpy as np
+import xarray as xr
+from affine import Affine
+
+import rioxarray
+from rioxarray.enum import Convention
+
+
+class TestConventionArchitecture:
```

Comment:
Mind using individual test functions instead of a class? The class isn't needed.